luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/KeyValueStoreDimensionFactory.java

```diff
@@ -0,0 +1,89 @@
+package com.yahoo.bard.webservice.config.luthier.factories;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.yahoo.bard.webservice.config.luthier.Factory;
+import com.yahoo.bard.webservice.config.luthier.LuthierIndustrialPark;
+import com.yahoo.bard.webservice.data.config.LuthierDimensionField;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+import com.yahoo.bard.webservice.data.dimension.KeyValueStore;
+import com.yahoo.bard.webservice.data.dimension.SearchProvider;
+import com.yahoo.bard.webservice.data.dimension.impl.KeyValueStoreDimension;
+import com.yahoo.bard.webservice.util.EnumUtils;
+
+import java.util.LinkedHashSet;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class KeyValueStoreDimensionFactory implements Factory<Dimension> {
+
+
+    /**
+     * @param fieldsNode the JsonNode object that points to the content of "fields" key
+     *                   or "defaultFields" key
+     * @return constructed fields associated with a dimension. Contains camelName, description, and tags
+     */
+    private LinkedHashSet<DimensionField> fieldsBuilder(JsonNode fieldsNode) {
+        LinkedHashSet<DimensionField> dimensionFields = new LinkedHashSet<>();
+        for(JsonNode node : fieldsNode) {
+            List<String> tags = new ArrayList<>();
+            if (node.has("tags")) {
+                for (final JsonNode strNode : node.get("tags")) {
+                    tags.add( strNode.textValue() );
+                }
+            }
+            dimensionFields.add(new LuthierDimensionField(
+                    EnumUtils.camelCase(node.get("name").textValue()),
+                    "Error: currently there is no description",             // TODO: Magic values!
```

Comment:
I would just use the name of the field here for now, or update the Lua stuff to include a descriptions
field for the dimension fields!

Reply:
Done. Used name for now. Populating each of the item in all `fields` is a little bit more hairy (because it involves more lua change). I will do it in the next PR, and definitely close this one ASAP!